packages/flutter/lib/src/material/floating_action_button_location.dart

```diff
@@ -12,53 +12,84 @@ import 'scaffold.dart';
 /// edge of the screen.
-/// 
+///
 /// [FloatingActionButtonLocation.endFloat] uses this to set the appropriate margin
 /// between the [FloatingActionButton] and the end of the screen.
-const double kFloatingActionButtonMargin = 16.0; 
+const double kFloatingActionButtonMargin = 16.0;
 
 /// The amount of time the [FloatingActionButton] takes to transition in or out.
-/// 
-/// The [Scaffold] uses this to set the duration of [FloatingActionButton] 
+///
+/// The [Scaffold] uses this to set the duration of [FloatingActionButton]
 /// motion, entrance, and exit animations.
 const Duration kFloatingActionButtonSegue = const Duration(milliseconds: 200);
 
 /// The fraction of a circle the [FloatingActionButton] should turn when it enters.
-/// 
+///
 /// Its value corresponds to 0.125 of a full circle, equivalent to 45 degrees or pi/4 radians.
 const double kFloatingActionButtonTurnInterval = 0.125;
 
 /// An object that defines a position for the [FloatingActionButton]
 /// based on the [Scaffold]'s [ScaffoldPrelayoutGeometry].
-/// 
+///
 /// Flutter provides [FloatingActionButtonLocation]s for the common
 /// [FloatingActionButton] placements in Material Design applications. These
 /// locations are available as static members of this class.
-/// 
+///
 /// See also:
-/// 
+///
 ///  * [FloatingActionButton], which is a circular button typically shown in the
 ///    bottom right corner of the app.
 ///  * [FloatingActionButtonAnimator], which is used to animate the
-///    [Scaffold.floatingActionButton] from one [FloatingActionButtonLocation] to 
+///    [Scaffold.floatingActionButton] from one [FloatingActionButtonLocation] to
 ///    another.
-///  * [ScaffoldPrelayoutGeometry], the geometry that 
+///  * [ScaffoldPrelayoutGeometry], the geometry that
 ///    [FloatingActionButtonLocation]s use to position the [FloatingActionButton].
 abstract class FloatingActionButtonLocation {
   /// Abstract const constructor. This constructor enables subclasses to provide
   /// const constructors so that they can be used in const expressions.
   const FloatingActionButtonLocation();
 
   /// End-aligned [FloatingActionButton], floating at the bottom of the screen.
-  /// 
+  ///
   /// This is the default alignment of [FloatingActionButton]s in Material applications.
   static const FloatingActionButtonLocation endFloat = const _EndFloatFabLocation();
 
   /// Centered [FloatingActionButton], floating at the bottom of the screen.
   static const FloatingActionButtonLocation centerFloat = const _CenterFloatFabLocation();
 
+  /// Locates the [FloatingActionButton] at the leading end of the
+  /// top of the app's content area, vertically centered over the border between
+  /// the [Scaffold.body] and the [Scaffold.appBar].
+  ///
+  /// This is unlikely to be a useful location for apps that lack an app bar.
+  static const FloatingActionButtonLocation startTop = const _StartTopFloatingActionButtonLocation();
```

Comment:
I think this particular location might be more useful left in the demo, unless it's a common location for placing the floating action button in Material apps.  I introduced it as a demonstration and not an implementation of any Material specification.

My reasoning is that even in apps that have an AppBar, this will cover the back button and the menu and title unless there's free space in the appbar.  This means that some amount of custom logic will be necessary, so someone putting a fab in the top start corner might need to create a custom location anyway.